src/actions/DeepLinkingActions.tsx

```diff
@@ -329,6 +330,30 @@ async function handleLink(navigation: NavigationBase, dispatch: Dispatch, state:
       break
     }
 
+    case 'rewards': {
+      const { pluginId, tokenId } = link
+
+      // Choose wallet:
+      const walletListResult = await pickWallet({
+        account,
+        assets: [{ pluginId, tokenId }],
+        navigation,
+        showCreateWallet: true
+      })
+      if (walletListResult?.type !== 'wallet') break
+      const { walletId } = walletListResult
+      const wallet = account.currencyWallets[walletId]
+      const { publicAddress } = (await wallet.getAddresses({ tokenId }))[0]
+      const { currencyCode } = tokenId == null ? account.currencyConfig[pluginId].currencyInfo : account.currencyConfig[pluginId].allTokens[tokenId]
```

Comment:
`account.currencyConfig[pluginId].allTokens[tokenId]` could technically be undefined so accessing `currencyCode` could cause a crash

Reply:
Instead of validating if the token exists here, maybe fix the broken enabledTokenIds lookup in pickWallet